site/docs/features/index.md

```diff
@@ -1,107 +1,327 @@
+    branch
+      |
+      |
+      v
++-----------+ 
+| commit #1 | 
++-----------+ 
+      ^
+      |
+      |
+   "master"
+    branch
+```
+Our example Spark job has two tasks, each generates a separate commit, which are only
+visible on our "work"-branch:
+```
+          task#1         task#2   "work"
+          result         result   branch
+            |                |     |
+            v                v     v
+      +-----------+       +-----------+
+      | commit #2 | --<-- | commit #3 |
+      +-----------+       +-----------+
+         |
+         v
+         |
++-----------+ 
+| commit #1 | 
++-----------+ 
+      ^
+      |
+      |
+   "master"
+    branch
+```
+When the job has finished, you can merge the now consistent result back
+into the "master"-branch.
 ```
+          task#1         task#2   "work"
+          result         result   branch
+            |                |     |
+            v                v     v
+      +-----------+       +-----------+
+      | commit #2 | --<-- | commit #3 |
+      +-----------+       +-----------+
+         |                          |        
+         v                          ^
+         |                          |
++-----------+                     +-----------+
+| commit #1 | --------<---------- | commit #4 |  
++-----------+                     +-----------+
+                                      ^
+                                      |
+                                      |
+                                   "master"
+                                    branch
+```
+Note that the merge-`commit #4` has references to two commits:
+
+* the previous commit on the "master" branch and
+* the merged commit from the "work"-branch
+
+It would be nice to give that `commit #4` a meaningful commit message, like
+`aggregate-financial-stuff 2020/12/24`, though, so people that look through the
+history of the data can grasp what that commit changes and why it's there.
+
+Before you actually merge the data from a working-branch into the "master" branch,
+it is possible to review all the changes made by your job - either programmatically,
+like running some process that validates the state of the branch, or asking a human
+to take a look.
+
+### Working branches for "humans"
+
+You can also use "developer" branches to run experiments against your data, test
+changes of your jobs, etc etc etc.
+
+Production and staging and development can use the same data lake without risking
+the consistent state of production data.
+
+### Squashing commits
+
+Nessie can also help to reduce the amount of data files in a data lake.
+
+You can for example merge all data files produces by all tasks of an analytics job
+before the whole change is actually merged into the "master" branch. Nessie will
+[figure out](#garbage-collection) that only the merged files need to survive and
+remove the unused files.
+
+### Tags
+
+Another type of named references are *tags*. Nessie *tags* are named references
```

Comment:
perhaps a comparison/link to git tags. Also: 'do not get automatically "moved".' made me think first that tags can be committed to but the ref wont follow the tag. Wonder if there is another way to express how tags work

Reply:
Added the link and re-phrased.